pkg/admission/clusterworkspacetypeexists/admission.go

```diff
@@ -156,6 +154,35 @@ func (o *clusterWorkspaceTypeExists) Admit(ctx context.Context, a admission.Attr
 	return updateUnstructured(u, cw)
 }
 
+func (o *clusterWorkspaceTypeExists) resolveClusterWorkspaceType(clusterName logicalcluster.Name, clusterWorkspaceType string) (*tenancyv1alpha1.ClusterWorkspaceType, bool, error) {
+	notFound := fmt.Errorf("spec.type %q does not exist", clusterWorkspaceType)
+	var cwt *tenancyv1alpha1.ClusterWorkspaceType
+	for {
+		parent, isChild := clusterName.Parent()
+		var getErr error
+		cwt, getErr = o.typeLister.Get(clusters.ToClusterAwareKey(clusterName, strings.ToLower(clusterWorkspaceType)))
+		if getErr != nil && !apierrors.IsNotFound(getErr) {
+			// something went wrong, fail this request
+			return nil, false, getErr
+		}
+		if apierrors.IsNotFound(getErr) && !isChild {
+			// we've traversed the full workspace hierarchy and could not find the type, fail
+			if clusterWorkspaceType == "Universal" {
+				return nil, true, nil // except that Universal is always valid
+			}
+			return nil, false, notFound
+		}
+		if getErr == nil || !isChild {
+			break
+		}
```

Comment:
a little complicated boolean case distinction here. Why not move the whole `!isChild` case to line 180? Now it looks like we have twice the same case?!